shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/keygen/SnowflakeKeyGenerateAlgorithm.java

```diff
@@ -96,13 +96,15 @@
     
     @Override
     public void init() {
-        workerId = getWorkerId();
         maxVibrationOffset = getMaxVibrationOffset();
         maxTolerateTimeDifferenceMilliseconds = getMaxTolerateTimeDifferenceMilliseconds();
     }
     
     private long getWorkerId() {
-        long result = Long.parseLong(props.getOrDefault(WORKER_ID_KEY, WORKER_ID).toString());
+        if (null == instanceContext) {
+            return DEFAULT_WORKER_ID;
+        }
+        long result = instanceContext.getWorkerId();
```

Comment:
@menghaoranss Why not init workerId in `init` method?

Reply:
@strongduanmu the workerId should be able to updated online, it's better to get it when need to use, and if workerId has been generated before, it will be retrieved directly from memory.

Reply:
BTW, thanks for you review

Reply:
> s for you review

@menghaoranss Thank you for your explanation.